trackio/deploy.py

```diff
@@ -27,25 +28,19 @@ def deploy_as_space(
     trackio_path = files("trackio")
 
     hf_api = huggingface_hub.HfApi()
-    whoami = None
-    login = False
+
     try:
-        whoami = hf_api.whoami()
-        if whoami["auth"]["accessToken"]["role"] != "write":
-            login = True
-    except OSError:
-        login = True
-    if login:
-        print("Need 'write' access token to create a Spaces repo.")
-        huggingface_hub.login(add_to_git_credential=False)
-        whoami = hf_api.whoami()
-
-    huggingface_hub.create_repo(
-        space_id,
-        space_sdk="gradio",
-        repo_type="space",
-        exist_ok=True,
-    )
+        huggingface_hub.create_repo(
+            space_id,
+            space_sdk="gradio",
+            repo_type="space",
+            exist_ok=True,
+        )
+    except HTTPError as e:
+        if e.response.status_code == 403:  # Forbidden
+            print("Need 'write' access token to create a Spaces repo.")
+            huggingface_hub.login(add_to_git_credential=False)
+        raise e
```

Comment:
I think this needs to catch 401s as well. If I'm not logged in locally, this raises a 401